WordPress/Sniffs/Theme/DeregisterScriptSniff.php

```diff
@@ -0,0 +1,43 @@
+<?php
+/**
+ * Sniff to check theme for wp_deregister_style()
+ *
+ * @category Theme
+ * @package  PHP_CodeSniffer
+ * @author   Simon Prosser <pross@pross.org.uk>
+ */
+class WordPress_Sniffs_Theme_DeregisterScriptSniff extends WordPress_Sniffs_Functions_FunctionRestrictionsSniff
+{
+
+	/**
+	 * Exclude groups
+	 *
+	 * Example: 'switch_to_blog,user_meta'
+	 *
+	 * @var string Comma-delimited group list
+	 */
+	public $exclude = '';
+
+	/**
+	 * Groups of functions to restrict
+	 *
+	 * Example: groups => array(
+	 * 	'lambda' => array(
+	 * 		'type' => 'error' | 'warning',
+	 * 		'message' => 'Use anonymous functions instead please!',
+	 * 		'functions' => array( 'eval', 'create_function' ),
+	 * 	)
+	 * )
+	 *
+	 * @return array
+	 */
+	public function getGroups() {
+		return array(
+			'wp_deregister_script' => array(
+				'type'	=> 'error',
```

Comment:
Same reason as why they try to do it with core scripts.
Example use-case: say a popular plugin uses an out-of-date version of a popular js library which is not included in core. The theme also uses this library but needs a more up to date version. They've tested with the popular plugin and found that the plugin is compatible with the newer version of the js lib, so they deregister the older version of the js library and register the newer instead using the same handle so as to make sure that there won't be conflicting js libs running at the same time.